notes-up fails to build

**Describe the bug**


**To Reproduce**
Steps to reproduce the behavior:
```
[davidak@gaming:~/code/nixpkgs]$ nix run -f channel:nixos-unstable notes-up
builder for '/nix/store/i2m6a38g9d2jya35ndsvdiwcqvbanfnr-notes-up-2.0.2.drv' failed with exit code 1; last 10 log lines:
                                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  /build/source/src/Services/WordWrapper.vala:34.48-34.96: warning: unhandled error `GLib.RegexError'
          WordWrapper.ending_whitespaces_regex = new Regex ("(\\s)+$", RegexCompileFlags.EXTENDED);
                                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  /build/source/src/GranitePatches/SourceList.vala:1516.17-1516.33: warning: `Granite.Widgets.Utils.set_theming' has been deprecated since 5.5.0. Use 
  /build/source/src/GranitePatches/SourceList.vala:1516.62-1516.91: warning: `Granite.StyleClass.SOURCE_LIST' has been deprecated since 0.4.2. Use Granite.STYLE_CLASS_SOURCE_LIST
  /build/source/src/Widgets/Editor.vala:311.9-311.31: warning: `Gtk.Widget.override_font' has been deprecated since 3.16
  Command-line option `--thread` is deprecated and will be ignored
  Compilation failed: 1 error(s), 18 warning(s)
  ninja: build stopped: subcommand failed.
[0 built (1 failed), 16.5 MiB DL]
error: build of '/nix/store/i2m6a38g9d2jya35ndsvdiwcqvbanfnr-notes-up-2.0.2.drv' failed
```

**Expected behavior**
no fail

**Additional context**
it seem to be fixed upstream, but no new release it made yet https://github.com/Philip-Scott/Notes-up/issues/361

**Notify maintainers**
@worldofpeace 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.8, NixOS, 21.05pre279012.ad47284f8b0 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"home-manager-20.09, nixos-21.05pre281004.3d1a7716d7f, nixos-hardware, nixos-unstable-21.05pre281004.3d1a7716d7f"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/applications/office/notes-up/default.nix
+++ b/pkgs/applications/office/notes-up/default.nix
@@ -1,9 +1,10 @@
-{ lib, stdenv
+{ lib
+, stdenv
 , fetchFromGitHub
 , nix-update-script
 , pantheon
 , pkg-config
-, vala_0_46
+, vala
 , cmake
 , ninja
 , gtk3
@@ -12,27 +13,28 @@
 , gtkspell3
 , glib
 , libgee
+, pcre
 , sqlite
 , discount
 , wrapGAppsHook
-, withPantheon ? false }:
+, withPantheon ? false
+}:
 
 stdenv.mkDerivation rec {
   pname = "notes-up";
-  version = "2.0.2";
+  version = "unstable-2020-12-29";
 
   src = fetchFromGitHub {
     owner = "Philip-Scott";
     repo = "Notes-up";
-    rev = version;
-    sha256 = "0bklgp8qrrj9y5m77xqbpy1ld2d9ya3rlxklgzx3alffq5312i4s";
+    rev = "2ea9f35f588769758f5d2d4436d71c4059141a6f";
+    sha256 = "sha256-lKOM9+s34xYB9bF9pgip9DFu+6AaxSE4HjFVhoWtttk=";
   };
 
   nativeBuildInputs = [
     cmake
     ninja
-    # fails with newer vala: https://github.com/Philip-Scott/Notes-up/issues/349
-    vala_0_46
+    vala
     pkg-config
     wrapGAppsHook
   ];
@@ -45,6 +47,7 @@ stdenv.mkDerivation rec {
     gtkspell3
     libgee
     pantheon.granite
+    pcre
     sqlite
     webkitgtk
   ];
@@ -60,9 +63,9 @@ stdenv.mkDerivation rec {
 
   meta = with lib; {
     description = "Markdown notes editor and manager designed for elementary OS"
-    + lib.optionalString withPantheon " - built with Contractor support";
+      + lib.optionalString withPantheon " - built with Contractor support";
     homepage = "https://github.com/Philip-Scott/Notes-up";
-    license = licenses.gpl2;
+    license = licenses.gpl2Only;
     maintainers = with maintainers; [ davidak ];
     platforms = platforms.linux;
   };

